samtranslator/translator/translator.py

```diff
@@ -152,7 +153,10 @@ def translate(self, sam_template, parameter_values, feature_toggle=None):
                 del template["Resources"][logical_id]
                 for resource in translated:
                     if verify_unique_logical_id(resource, sam_template["Resources"]):
-                        template["Resources"].update(resource.to_dict())
+                        _r = resource.to_dict()
+                        if resource_dict.get("Metadata") and convert_metadata:
```

Comment:
I think a comment here could go a long away in understanding what is happening. It took me a while but I think what is happening is the following:
For each SAM resource in the source template, we translate to the underlying CloudFormation Resources. After we translate, we append any metadata in the SAM Resource to the translated CloudFormation Resources.

They generally works but I think there might be a couple edge cases outstanding:
1) S3 buckets are updated (I made a comment below about this)
2) I am not sure how generated resources work in this case. If you have an API Event on an `AWS::Serverless::Function.Event` what happens? Does the Metadata get pass through correctly? That is the `Metadata` on `AWS::Serverless::Function` gets added to all the API Gateway resources? This would be a good test case to add regardless of it it works in the current implementation or not.

Reply:
1. see above

2. Yes. A single SAM resource translates to multiple CFN resources. The metadata on the single SAM resource is passed through to the generated CFN resources, regardless of what the resource type is.